gluoncv/model_zoo/efficientnet.py

```diff
@@ -0,0 +1,508 @@
+            with self.features.name_scope():
+                # stem conv
+                out_channels = round_filters(32, self._global_params)
+                _add_conv(
+                    self.features,
+                    out_channels,
+                    kernel=3,
+                    stride=2,
+                    active=True,
+                    batchnorm=True)
+            self._blocks = nn.HybridSequential(prefix='blocks_')
+            with self._blocks.name_scope():
+                for block_arg in self._blocks_args:
+                    # Update block input and output filters based on depth
+                    # multiplier.
+                    block_arg = block_arg._replace(
+                        input_filters=round_filters(
+                            block_arg.input_filters, self._global_params),
+                        output_filters=round_filters(
+                            block_arg.output_filters, self._global_params),
+                        num_repeat=round_repeats(
+                            block_arg.num_repeat, self._global_params))
+                    self._blocks.add(MBConv(block_arg.input_filters,
+                                            block_arg.output_filters,
+                                            block_arg.expand_ratio,
+                                            block_arg.kernel_size,
+                                            block_arg.stride,
+                                            block_arg.se_ratio,
+                                            global_params.drop_connect_rate)
+                                     )
+                    if block_arg.num_repeat > 1:
+                        block_arg = block_arg._replace(
+                            input_filters=block_arg.output_filters, stride=1)
+                    for _ in range(block_arg.num_repeat - 1):
+                        self._blocks.add(
+                            MBConv(
+                                block_arg.input_filters,
+                                block_arg.output_filters,
+                                block_arg.expand_ratio,
+                                block_arg.kernel_size,
+                                block_arg.stride,
+                                block_arg.se_ratio,
+                                global_params.drop_connect_rate))
+
+            # Head
+            out_channels = round_filters(1280, self._global_params)
+            self._conv_head = nn.HybridSequential(prefix='conv_head_')
+            with self._conv_head.name_scope():
+                _add_conv(
+                    self._conv_head,
+                    out_channels,
+                    active=True,
+                    batchnorm=True)
+            # Final linear layer
+            self._dropout = self._global_params.dropout_rate
+            self._fc = nn.Dense(self._global_params.num_classes)
+
+    def hybrid_forward(self, F, x, training):
```

Comment:
why do we need an additional "training" here?

Reply:
By using `autograd.is_training()` , the implementation can get rid of the `training` argument.